beps/0003-auth-architecture-evolution/README.md

```diff
@@ -130,96 +212,91 @@ export interface HttpAuthService {
-  },
-);
+router.get('/read-data', (req, res) => {
+  // The catalogClient will have a reference to the (plugin scoped) HttpAuthService,
+  // which it uses to create the credential headers for the upstream request.
+  const entity = await catalogClient.getEntityByRef(req.params.entityRef, {
+    credentials: httpAuth.credentials(req),
+  });
+  // ...
+});
 ```
 
-**Allow both user and service request**
+#### Allow both user and service request
 
 ```ts
-router.get(
-  '/read-data',
-  httpAuth.middleware({ allow: ['user', 'service'] }),
-  (req, res) => {
-    const credentials = httpAuth.credentials(req);
-    if (credentials.user) {
-      res.json(
-        // Silly example just to highlight separate code paths for user and
-        // service requests
-        todoStore.listOwnedTodos({ owner: credentials.user.userEntityRef }),
-      );
-    } else {
-      res.json(todoStore.listTodos());
-    }
-  },
-);
+router.get('/read-data', (req, res) => {
+  const credentials = httpAuth.credentials(req);
+  if (credentials.user) {
+    res.json(
+      // Silly example just to highlight separate code paths for user and
+      // service requests
+      todoStore.listOwnedTodos({ owner: credentials.user.userEntityRef }),
+    );
+  } else {
+    res.json(todoStore.listTodos());
+  }
+});
 ```
 
-**Issuing a cookie and allowing user cookie auth on a separate endpoint**
+#### Issuing a cookie and allowing user cookie auth on a separate endpoint
 
 ```ts
-router.get('/cookie', httpAuth.middleware({ allow: ['user'] }), (req, res) => {
-  await httpAuth.issueUserCookie(res);
+router.get('/cookie', (req, res) => {
+  await httpAuth.issueUserCookie(res); // If this is a service call it'll throw
   res.json({ ok: true });
 });
 
+// Allowing cookie auth is a separate step where you call the configure method
+// of the httpRouter API in your plugin setup code.
+httpRouter.configure({
+  // In practice we can make this configuration a lot more capable, this is just a minimal example
+  allowCookieAuthOnPaths: ['/static'], // router.use('/static', cookieAuthMiddleware()) under the hood
+});
+
 // Separate endpoint that serves static content, allowing user cookie auth as
-// well as regular user auth
-router.use(
-  '/static',
-  httpAuth.middleware({ allow: ['user', 'user-cookie'] }),
-  express.static(staticContentDir),
-);
+// well as the default user and service auth methods
+router.use('/static', express.static(staticContentDir));
 ```
 
-**Passing along user identity from a cookie in an upstream request**
+#### Passing along user identity from a cookie in an upstream request
 
 ```ts
 router.get(
   '/read-data',
   httpAuth.middleware({ allow: ['user-cookie'] }),
   (req, res) => {
-    const { cookieUser } = httpAuth.credentials(req);
+    // These credentials don't actually contain an underlying user token for cookie-authed requests
+    // You you try to pass them to the AuthService, it'll throw.
+    const { user } = httpAuth.credentials(req);
```

Comment:
do these need to await too? or can we be sure that ALL async work (verification etc) will have happened in the surrounding middleware?

Reply:
Should be possible to ensure that it's sync, thinking that can help access patterns quite a lot